app/models/foreman_snapshot_management/snapshot.rb

```diff
@@ -37,17 +34,8 @@ def self.new_from_vmware(host, raw_snapshot, opts = {})
       )
     end
 
-    def children
-      return [] unless raw_snapshot
-
-      child_snapshots = raw_snapshot.child_snapshots.flat_map do |child_snapshot|
-        self.class.new_from_vmware(host, child_snapshot, parent: self)
-      end
-      child_snapshots + child_snapshots.flat_map(&:children)
-    end
```

Comment:
@timogoebel Do you know, whether `children` is used anywhere?
Here we do not anymore since there is `recursive: true`.

Reply:
Isn't it used in the API response?

I'm having a hard time figuring out what you want to achieve with this refactoring. Ideally, we'd drop any extensions to fog-vsphere completely.

Reply:
I thought about how to add support for other compute_resources and realized, that we do a lot of vmware specific hacks in snapshot.rb, so i am trying to move them into vmware_extension concern. And i think, this parental relation of snapshots is unique to vmware.
I think, that concern will stay necessary, because the snapshot interface in fog does not seem to be consistent across different models.

Reply:
I think it'd be better to subclass `Snapshot` for different providers. Ideally, we'd move all fog specific code to the for repos. Monkey patching works, but is very hard to maintain.

Reply:
For getting the snapshot capability into vmware compute resource, we'd either need to do the concern, or move the whole functionality to foreman source, afaik.
What would you think about that action? We could circumvent using deface (which is monkey patching on the http level), also.

Reply:
I think the concern is fine, I meant extensions to `fog-vsphere`.

Reply:
We can get rid of deface using this pattern:
https://github.com/theforeman/foreman_omaha/blob/7d6b9cf87582f7ba7f103b1504a14f95ed04e1e8/lib/foreman_omaha/engine.rb#L82-L88